cmd/helm/repo_add_test.go

```diff
@@ -86,3 +88,51 @@ func TestRepoAdd(t *testing.T) {
 		t.Errorf("Duplicate repository name was added")
 	}
 }
+
+func TestRepoAddConcurrentGoRoutines(t *testing.T) {
+	const testName = "test-name"
+
+	ts, err := repotest.NewTempServer("testdata/testserver/*.*")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ts.Stop()
+
+	repoFile := filepath.Join(ensure.TempDir(t), "repositories.yaml")
+
+	var wg sync.WaitGroup
+	wg.Add(3)
+	for i := 0; i < 3; i++ {
+		go func(name string) {
+			defer wg.Done()
+			o := &repoAddOptions{
+				name:     name,
+				url:      ts.URL(),
+				noUpdate: true,
+				repoFile: repoFile,
+			}
+			if err := o.run(ioutil.Discard); err != nil {
+				t.Error(err)
+			}
+		}(fmt.Sprintf("%s-%d", testName, i))
+	}
+	wg.Wait()
+
+	b, err := ioutil.ReadFile(repoFile)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var f repo.File
+	if err := yaml.Unmarshal(b, &f); err != nil {
+		t.Error(err)
+	}
+
+	var name string
+	for i := 0; i < 3; i++ {
+		name = fmt.Sprintf("%s-%d", testName, i)
+		if !f.Has(name) {
+			t.Errorf("%s was not successfully inserted into %s: %s", name, repoFile, f.Repositories[0])
+		}
+	}
+}
```

Comment:
Did we want to pull over the [subprocess test](https://github.com/helm/helm/pull/5678/files#diff-5537931905cc99a35fc6d296d4e3feadR154) as well? I figure this concurrency locking will most likely occur with multiple processes

Reply:
The subprocess test may likely be the cause of the circleci failures in helm 2. It's excessive, flaky, and unnecessary IMO. The parallel waitgroup test should be more than enough to cover the case

Reply:
👌